rllib/core/learner/learner_group.py

```diff
@@ -474,46 +458,101 @@ def get_weights(self, module_ids: Optional[Set[str]] = None) -> Mapping[str, Any
                 lambda w: w.get_state(), remote_actor_ids=[worker]
             )
-            return self._get_results(results)[0]
+            learner_state = self._get_results(results)[0]
+        return {
+            "learner_state": learner_state,
+            "should_module_be_updated_fn": self.should_module_be_updated_fn,
+        }
 
-    def set_state(self, state: List[Mapping[ModuleID, Mapping[str, Any]]]) -> None:
-        """Sets the states of the Learners.
+    def set_state(self, state: Mapping[str, Any]) -> None:
+        """Sets the state of this LearnerGroup.
 
-        Args:
-            state: The state of the Learners
+        Note that all Learners share the same state.
 
+        Args:
+            state: The state dict mapping str keys to state information.
         """
-        if self.is_local:
-            self._learner.set_state(state)
-        else:
-            self._worker_manager.foreach_actor(lambda w: w.set_state(state))
+        learner_state = state.get("learner_state")
+        if learner_state is not None:
+            if self.is_local:
+                self._learner.set_state(learner_state)
+            else:
+                self._worker_manager.foreach_actor(lambda w: w.set_state(learner_state))
+        if state.get("should_module_be_updated_fn"):
+            self.set_should_module_be_updated_fn(state["should_module_be_updated_fn"])
 
-    def set_is_module_trainable(
-        self, is_module_trainable: Callable[[ModuleID, MultiAgentBatch], bool] = None
+    @property
+    def should_module_be_updated_fn(self):
+        return self._should_module_be_updated_fn
+
+    def set_should_module_be_updated_fn(
+        self, should_module_be_updated_fn: Optional[ShouldModuleBeUpdatedFn] = None
     ) -> None:
-        """Sets the function that determines whether a module is trainable.
+        """Sets the function that determines whether a module should be updated or not.
 
         Args:
-            is_module_trainable: A function that takes in a module id and a batch
-                and returns a boolean indicating whether the module should be trained
-                on the batch.
+            should_module_be_updated_fn: An optional callable that takes in a ModuleID
+                and a batch and returns a boolean indicating whether the module should
+                be updated on the given batch.
         """
-        if is_module_trainable is not None:
-            self._is_module_trainable = is_module_trainable
+        # If None, use default implementation (all modules should be updated).
+        if should_module_be_updated_fn is None:
+            should_module_be_updated_fn = _default_should_module_be_updated_fn
+        # If container given, construct a simple callable returning True
+        # if the ModuleID is found in the list/set of IDs.
+        elif not callable(should_module_be_updated_fn):
+            assert isinstance(should_module_be_updated_fn, (list, set, tuple)), (
```

Comment:
don't raise assertion errors for things that user have to take an action on. Only use assertion for internal violation of assumption, things that should point to a bug. When people report an assertion error we immediately know we have a bug an not mis-use.

Reply:
You are right! Fixed.